esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -1543,6 +1549,99 @@ bool CWsDfuEx::onDFUArrayAction(IEspContext &context, IEspDFUArrayActionRequest
     return true;
 }
 
+bool CWsDfuEx::changeFileProtections(IEspContext &context, IEspDFUArrayActionRequest &req, IEspDFUArrayActionResponse &resp)
+{
+    CDFUChangeProtection protect = req.getProtect();
+    if (protect == CDFUChangeProtection_NoChange)
+        return true;
+
+#ifdef _USE_OPENLDAP
+    if (protect == CDFUChangeProtection_UnprotectAll)
+        context.ensureSuperUser(ECLWATCH_SUPER_USER_ACCESS_DENIED, "Access denied, administrators only.");
+#endif
+
+    if (isDetachedFromDali())
+        throw makeStringException(ECLWATCH_INVALID_INPUT, "ESP server is detached from Dali. Please try later.");
+
+    StringBuffer userID;
+    context.getUserID(userID);
+    Owned<IUserDescriptor> userDesc;
+    if (userID.isEmpty())
+        userID.set("hpcc"); //The userID is needed for setProtect().
```

Comment:
@wangkx under what circumstances does it get here?  Only when running with no security?

Reply:
Yes, only when running with no security.